src/assets/scripts/client/navigationLibrary/NavigationLibrary.js

```diff
@@ -0,0 +1,93 @@
+import PositionModel from '../base/PositionModel';
+import FixCollection from './Fix/FixCollection';
+import StandardRouteCollection from './StandardRoute/StandardRouteCollection';
+
+/**
+ * @class NavigationLibrary
+ */
+export default class NavigationLibrary {
+    /**
+     * @constructor
+     * @for NavigationLibrary
+     * @param airportJson {object}
+     */
+    constructor(airportJson) {
+        /**
+         *
+         *
+         * @property _referencePosition
+         * @type {PositionModel}
+         * @default null
+         */
+        this._referencePosition = null
+
+        /**
+         *
+         *
+         * @property _sidCollection
+         * @type {StandardRoute}
+         * @default null
+         */
+        this._sidCollection = null;
+
+        /**
+         *
+         *
+         * @property _starCollection
+         * @type {StandardRoute}
+         * @default null
+         */
+        this._starCollection = null;
```

Comment:
Yes and no. In cases like youve described, depending on which airport you go to you could have slightly different routings at the end to get you pointed toward the right airport. That could probably be addressed by adding more sections within the STAR, but could get complicated. Or, take SIDs: how would we deal with the runway sections? Airport A and Airport B both use SID "URSLA4". Airport A has runway 5-23, and airport B has runways 1L-19R, 1R-19L, and 5-23. Each runway has some particulars to it. The logistics of distinguishing between these sorts of things when the SID is just in the nav library would be pretty clunky I would tend to think. I may be able to find a departure that has that, if you like. At least one example (not common runway names) is the MIA5 SID off KMIA and KTMB, a Miami satellite airport. Same Sid, different initial instructions.